lib/class-wp-rest-media-search-handler-gutenberg.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * REST API: WP_REST_Media_Search_Handler class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since X.X.X
+ */
+
+if ( class_exists( 'WP_REST_Media_Search_Handler' ) ) {
+	return;
+}
+
+/**
+ * Core class representing a search handler for attachments in the REST API.
+ *
+ * @since X.X.X
+ *
+ * @see WP_REST_Search_Handler
+ */
+class WP_REST_Media_Search_Handler_Gutenberg extends WP_REST_Post_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since X.X.X
+	 */
+	public function __construct() {
+		parent::__construct();
+		$this->type     = 'media';
+		$this->subtypes = array();
+	}
+
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @since X.X.X
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
+
+		$query_args = array(
+			'post_type'      => 'attachment',
```

Comment:
I'm looking at the original core function to compare; why are we assuming here that the `post_type` will always be an `attachment`?

Reply:
Same for `ignore_sticky_posts`